File: ios/Oriveo/Oriveo/Features/Home/HomeNotesEntryCard.swift
import SwiftUI

/// The fixed Notes entry card on Home, built from the same material as the hero composer card
/// (AuroraHeroSurface with a single soft glow top-right).
///
/// Left: `Notes` 17/semibold, the count (monospaced accent, baseline aligned) and the title of the
/// most recent note (single line, truncated).
/// Right: the Lucide notebook-pen line glyph at 40pt (without the binder ticks, see notebookGlyph),
/// tinted with a #C4B5FD → #EC8FEA → #8DB4FF linear gradient.
/// No chevron, no count badge, no watermark, no solid icon tile; the whole card opens the notes list.
struct HomeNotesEntryCard: View {
    let count: Int
    /// Title of the most recently updated note (`noteSummaries` sorted by updatedAt DESC, `.first`).
    /// nil when there are no notes.
    let latestTitle: String?
    let onOpen: () -> Void

    @Environment(\.colorScheme) private var colorScheme

    private var hasNotes: Bool { count > 0 }

    private static let cornerRadius: CGFloat = 26

    var body: some View {
        let shape = RoundedRectangle(cornerRadius: Self.cornerRadius, style: .continuous)

        Button(action: onOpen) {
            HStack(spacing: 16) {
                VStack(alignment: .leading, spacing: 4) {
                    HStack(alignment: .firstTextBaseline, spacing: 8) {
                        Text(L10n.tr("Notes", table: .notes))
                            .font(.system(size: 17, weight: .semibold))
                            .tracking(-0.3)
                            .foregroundStyle(AuroraTheme.Colors.textPrimary)
                            .lineLimit(1)
                        if hasNotes {
                            Text("\(count)")
                                .font(AuroraTheme.Typography.countMono)
                                .foregroundStyle(AuroraTheme.Colors.accent)
                        }
                    }
                    // Line heights are 22 / 18 (SF's natural line heights are about 20 / 15.3);
                    // without padding the line boxes the card ends up 4pt short.
                    .frame(minHeight: 22)

                    Text(previewLine)
                        .font(.system(size: 13))
                        .foregroundStyle(AuroraTheme.Colors.textSecondary)
                        // With notes: the latest title on one truncated line. Empty state: the invitation
                        // may take two lines, half a sentence would not be readable.
                        .lineLimit(hasNotes ? 1 : 2)
                        .truncationMode(.tail)
                        .multilineTextAlignment(.leading)
                        .fixedSize(horizontal: false, vertical: true)
                        .frame(minHeight: 18, alignment: .leading)
                }
                .frame(maxWidth: .infinity, alignment: .leading)

                notebookGlyph
            }
            // A 1px border plus padding 16/18/16/20 (border-box): content sits 1pt further from each edge.
            .padding(.top, 17)
            .padding(.trailing, 19)
            .padding(.bottom, 17)
            .padding(.leading, 21)
            .background(AuroraHeroSurface(cornerRadius: Self.cornerRadius, glow: .notesCorner))
            // In dark mode the same aurora crown as the hero, one step weaker, without the near-white hot
            // core and without the halo outside the card: Notes does not compete with the hero.
            .overlay {
                if colorScheme == .dark {
                    AuroraCrownRim(cornerRadius: Self.cornerRadius, scale: AuroraCrown.notesRimScale, hotCore: false)
                } else {
                    shape.strokeBorder(AuroraHeroSurface.rim(isDark: false), lineWidth: 1)
                }
            }
            .contentShape(shape)
        }
        .buttonStyle(.plain)
        .accessibilityLabel(accessibilityText)
    }

    /// The notebook line glyph: the NotesNotebookLine template vector (stroke 1.6 on a 24 grid).
    /// The gradient is defined in objectBoundingBox terms, so the zero-height binder ticks would never
    /// pick up colour and the body is filled across its own bounding box (4,2)→(20,22). The asset
    /// therefore omits the ticks, and the gradient endpoints follow the body's bounding box rather
    /// than the full 24 grid.
    private var notebookGlyph: some View {
        Image("NotesNotebookLine")
            .renderingMode(.template)
            .resizable()
            .aspectRatio(contentMode: .fit)
            .frame(width: 40, height: 40)
            .foregroundStyle(
                LinearGradient(
                    stops: [
                        .init(color: Color(hex: 0xC4B5FD), location: 0),
                        .init(color: Color(hex: 0xEC8FEA), location: 0.5),
                        .init(color: Color(hex: 0x8DB4FF), location: 1),
                    ],
                    startPoint: UnitPoint(x: 4.0 / 24, y: 2.0 / 24),
                    endPoint: UnitPoint(x: 20.0 / 24, y: 22.0 / 24)
                )
            )
            .accessibilityHidden(true)
    }

    private var previewLine: String {
        Self.previewLine(count: count, latestTitle: latestTitle)
    }

    private var accessibilityText: String {
        Self.accessibilityText(count: count, latestTitle: latestTitle)
    }

    /// With notes: the latest title. Empty state (or a latest note without a title): the invitation
    /// copy, reusing the existing key.
    static func previewLine(count: Int, latestTitle: String?) -> String {
        if count > 0, let latestTitle, !latestTitle.isEmpty {
            return latestTitle
        }
        return L10n.tr("Save strong answers with their source, then return to them later.", table: .notes)
    }

    /// When there are notes, VoiceOver also reads the latest title shown on the card, so the spoken and
    /// visible information match.
    static func accessibilityText(count: Int, latestTitle: String?) -> String {
        let title = L10n.tr("Notes", table: .notes)
        if count > 0 {
            let saved = title + ", " + String(format: L10n.tr("%d saved", table: .notes), count)
            guard let latestTitle, !latestTitle.isEmpty else { return saved }
            return saved + ", " + latestTitle
        }
        return title + ". " + L10n.tr("Save strong answers with their source, then return to them later.", table: .notes)
    }
}

File: android/app/src/main/java/ai/oriveo/community/feature/notes/HomeNotesEntryCard.kt
package ai.oriveo.community.feature.notes

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.pluralStringResource
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.semantics.clearAndSetSemantics
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.role
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import ai.oriveo.community.R
import ai.oriveo.community.feature.home.AuroraTheme
import ai.oriveo.community.feature.home.auroraNotesSurface
import ai.oriveo.community.ui.theme.OriveoTheme

private val HOME_NOTES_CORNER_RADIUS = 26.dp

/**
 * The fixed Notes entry card on Home (matches iOS HomeNotesEntryCard): the same material as the hero composer,
 * keeping only the top-right glow.
 *
 * Left: `Notes` 17/semibold + count (mono accent, baseline aligned) + the latest note title (single line);
 * right: a 40dp notebook-pen line glyph (without the binding ticks) in a #C4B5FD → #EC8FEA → #8DB4FF gradient.
 * No chevron, count badge, watermark or solid icon tile; the whole card opens the notes list.
 * The design is a 1dp stroke plus padding 16/18/16/20 (border-box), so content sits 1dp further from the outer
 * edge: 17 / 19 / 17 / 21.
 */
@Composable
fun HomeNotesEntryCard(
    noteCount: Int,
    latestTitle: String?,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val isDark = OriveoTheme.isDark
    val hasNotes = noteCount > 0
    val shape = RoundedCornerShape(HOME_NOTES_CORNER_RADIUS)
    val previewLine = homeNotesPreviewLine(
        count = noteCount,
        latestTitle = latestTitle,
        untitled = stringResource(R.string.notes_untitled),
        invitation = stringResource(R.string.notes_subtitle),
    )
    val title = stringResource(R.string.home_notes_title)
    val countText = if (hasNotes) pluralStringResource(R.plurals.notes_count, noteCount, noteCount) else null
    val accessibilityText = homeNotesAccessibilityText(title = title, countText = countText, previewLine = previewLine)

    Row(
        modifier = modifier
            .fillMaxWidth()
            .auroraNotesSurface(isDark = isDark, cornerRadius = HOME_NOTES_CORNER_RADIUS)
            .clip(shape)
            .clickable(onClick = onClick)
            // With notes, read the latest title shown on the card as well (TalkBack hears what is visible)
            .clearAndSetSemantics {
                contentDescription = accessibilityText
                role = Role.Button
            }
            .padding(start = 21.dp, top = 17.dp, end = 19.dp, bottom = 17.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(16.dp),
    ) {
        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(4.dp),
        ) {
            Row(
                // The design uses line heights 22 / 18; without padding the line boxes the card ends up too short
                modifier = Modifier.heightIn(min = 22.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Text(
                    text = title,
                    fontSize = 17.sp,
                    lineHeight = 22.sp,
                    fontWeight = FontWeight.SemiBold,
                    letterSpacing = (-0.3).sp,
                    color = AuroraTheme.textPrimary(),
                    maxLines = 1,
                    modifier = Modifier.alignByBaseline(),
                )
                if (hasNotes) {
                    Text(
                        text = "$noteCount",
                        style = AuroraTheme.Typography.countMono,
                        color = AuroraTheme.accent(),
                        maxLines = 1,
                        modifier = Modifier
                            .padding(start = 8.dp)
                            .alignByBaseline(),
                    )
                }
            }
            Text(
                text = previewLine,
                fontSize = 13.sp,
                lineHeight = 18.sp,
                color = AuroraTheme.textSecondary(),
                // With notes: the latest title on one line; empty: the invitation may wrap to two lines, half a sentence would be unreadable
                maxLines = if (hasNotes) 1 else 2,
                overflow = TextOverflow.Ellipsis,
                modifier = Modifier.heightIn(min = 18.dp),
            )
        }
        Icon(
            imageVector = NotesNotebookLineGlyph,
            contentDescription = null,
            modifier = Modifier.size(40.dp),
            tint = Color.Unspecified,
        )
    }
}

/** With notes → the latest title (an empty title after trimming shows "Untitled", matching iOS NoteText.displayTitle); empty → the invitation copy. */
internal fun homeNotesPreviewLine(count: Int, latestTitle: String?, untitled: String, invitation: String): String {
    if (count <= 0) return invitation
    return latestTitle?.trim().orEmpty().ifEmpty { untitled }
}

internal fun homeNotesAccessibilityText(title: String, countText: String?, previewLine: String): String =
    if (countText == null) "$title. $previewLine" else "$title, $countText, $previewLine"
